File: test-vulnerable-code.js
// Intentionally vulnerable code for testing PR scan workflow
// This file contains security issues that the SAST engine should detect

const express = require('express');
const app = express();

// SQL Injection via string concatenation (js-sql-string-concat)
function getUser(userId) {
  const query = "SELECT * FROM users WHERE id = " + userId;
  return db.query(query);
}

// XSS via innerHTML assignment (js-xss-innerhtml-assignment)
function renderContent(userInput) {
  document.getElementById('output').innerHTML = userInput;
}

// eval usage (js-eval-usage)
function processInput(data) {
  eval(data);
}

// document.write XSS (js-xss-document-write)
function writeOutput(content) {
  document.write(content);
}

// SQL template literal injection (js-sql-template-literal)
function deleteUser(id) {
  const query = `DELETE FROM users WHERE id = ${id}`;
  return db.query(query);
}

// dangerouslySetInnerHTML (js-xss-dangerously-set-inner-html)
function RenderHTML({ html }) {
  return <div dangerouslySetInnerHTML={{ __html: html }} />;
}

module.exports = { getUser, renderContent, processInput, writeOutput, deleteUser };

File: test-vulnerable-code.py
# Intentionally vulnerable Python code for testing PR scan workflow

import os
import subprocess
import pickle

# Command injection via os.system (py-os-system)
def run_command(user_input):
    os.system("ls " + user_input)

# Command injection via os.popen (py-os-popen)
def read_output(filename):
    os.popen("cat " + filename)

# Dangerous exec usage (py-exec-usage)
def execute_code(code):
    exec(code)

# Dangerous eval usage (py-eval-usage)
def evaluate(expression):
    return eval(expression)

# Unsafe deserialization (py-pickle-loads)
def load_data(data):
    return pickle.loads(data)

# subprocess with shell=True (py-subprocess-shell-true)
def run_shell(cmd):
    subprocess.run(cmd, shell=True)
